src/jsCodegen/emitCompare.ts

```diff
@@ -0,0 +1,248 @@
+import { SequenceMultiplicity, SequenceType, ValueType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+import { CodeGenContext } from './CodeGenContext';
+import { emitOperand } from './emitOperand';
+// import { emitBaseExpr } from './emitBaseExpression';
+import {
+	acceptAst,
+	CompiledResultType,
+	FunctionIdentifier,
+	GeneratedCodeBaseType,
+	getCompiledValueCode,
+	PartialCompilationResult,
+	rejectAst,
+} from './JavaScriptCompiledXPath';
+
+/**
+ * Generates javascript code for a value compare expression.
+ *
+ * @param ast The ast of the value compare expression
+ * @param compareType The type of comparison we're executing
+ * @param firstExpr The generated code of the first expression
+ * @param secondExpr The generated code of the second expression
+ * @param identifier The identifier of the result
+ * @param staticContext The code generation context
+ * @returns Generated code of the value compare
+ */
+export function emitValueCompare(
+	ast: IAST,
+	compareType: string,
+	firstExpr: PartialCompilationResult,
+	secondExpr: PartialCompilationResult,
+	identifier: string,
+	_staticContext: CodeGenContext
+): PartialCompilationResult {
+	const leftType = astHelper.getAttribute(
+		astHelper.getFirstChild(ast, 'firstOperand')[1] as IAST,
+		'type'
+	);
+	const rightType = astHelper.getAttribute(
+		astHelper.getFirstChild(ast, 'secondOperand')[1] as IAST,
+		'type'
+	);
+
+	if (!leftType || !rightType) {
+		return rejectAst('Left or right type of compare are not found, annotation failed.');
+	}
+
+	const supportedTypes = [ValueType.XSSTRING, ValueType.NODE];
+	if (!supportedTypes.includes(leftType.type) || !supportedTypes.includes(rightType.type)) {
+		return rejectAst('Unsupported types in compare');
+	}
+
+	if (!firstExpr.isAstAccepted || !secondExpr.isAstAccepted) {
+		return rejectAst("One of the two operands in compare wasn't accepted");
```

Comment:
or just return the first or second expression? That might be more helpful

Reply:
This has been fixed now